### PR TITLE
Add Page Modules concept for integration tests

### PR DESCRIPTION
**Pull request by jbranchaud**
This is an experiment in using Page Modules (ala Page Objects) for feature tests. What do y'all think?

### CODE REVIEW

**Comment by keathley** [comment]
These look really great! If you're using the most recent version then you can swap a lot of your calls to `Browser.find` to force the browser to block with `Browser.assert_has`. `assert_has` is chainable so you don't have to worry about returning the `session` explicitly.

**Comment by tmock12** [comment]
OMG I ❤️ this @jbranchaud 

After updating some deprecation warnings in the tests I realized they were too hard to parse to just get the general sense of what the test does. This would fix that. 👍 from me.

**Comment by jwworth** [comment]
Thank you! 👍
